app/controllers/sessions_controller.rb

```diff
@@ -1,7 +1,7 @@
 class SessionsController < ApplicationController
   def new
     origin = { origin: session.delete(:return_to) }.to_query
-    redirect_to("/auth/google_oauth2?#{origin}")
```

Comment:
Appending origin creates some issues on logout. The old OAuth state is returned and an OmniAuth OAuth2 error of CSRF detected is thrown.

Reply:
I confirmed this by removing `origin` from the `redirect_to`. This is probably an issue with my `redirectUri` configuration though.

Reply:
@olivierlacan do you have any objections to adding a landing page to the application? Appears that all pages require authentication and it's a little jarring.

Reply:
I'd like for that to be configurable in the future (public articles) but you're right that it's jarring, I think it's a good idea to tell users we need to redirect them.